crates/biome_js_analyze/src/lint/nursery/use_readonly_class_properties.rs

```diff
@@ -0,0 +1,517 @@
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### checkAllProperties
+    ///
+    /// Check on all properties (`public` and `protected` properties). Default: false.
+    ///
+    pub UseReadonlyClassProperties {
+        version: "1.0.0",
+        name: "useReadonlyClassProperties",
+        language: "ts",
+        sources: &[RuleSource::EslintTypeScript("prefer-readonly")],
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+declare_node_union! {
+    pub AnyClassProperties = JsPropertyClassMember | TsPropertyParameter
+}
+
+impl Rule for UseReadonlyClassProperties {
+    type Query = Ast<JsClassDeclaration>;
+    type State = AnyClassProperties;
+    type Signals = Box<[Self::State]>;
+    type Options = ReadonlyClassPropertiesOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let class_declaration = ctx.query();
+        let options = ctx.options();
+        let eligible_properties =
+            get_eligible_properties(class_declaration, !options.check_all_properties);
+        if eligible_properties.is_empty() {
+            Vec::new()
+        } else {
+            find_properties_need_add_readonly(class_declaration.syntax(), eligible_properties)
+        }
+        .into_boxed_slice()
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let mut name = String::new();
+        if let Some(token) = state.text_trimmed() {
+            name = token;
+        }
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            state.range(),
+            markup! {
+                "Member '"{name}"' is never reassigned; mark it as `readonly`."
+            },
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let mut mutation = ctx.root().begin();
+        mutation.replace_element_discard_trivia(state.into_syntax(), state.replace_syntax());
+
+        Some(JsRuleAction::new(
+            ctx.metadata().action_category(ctx.category(), ctx.group()),
+            ctx.metadata().applicability(),
+            markup! { "Add `readonly` decorator." }.to_owned(),
+            mutation,
+        ))
+    }
+}
+
+/// Rule's options.
+#[derive(
+    Default, Debug, Clone, Deserializable, Eq, PartialEq, serde::Deserialize, serde::Serialize,
+)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct ReadonlyClassPropertiesOptions {
+    /// If `true`, then check all class properties member
+    ///
+    /// If `false`, then only check private members
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub check_all_properties: bool,
+}
+
+fn is_default<T: Default + Eq>(value: &T) -> bool {
+    value == &T::default()
+}
+
+fn get_constructor_eligible_params(
+    class_declaration: &JsClassDeclaration,
+    only_private: bool,
+) -> FxHashSet<AnyClassProperties> {
```

Comment:
Why not just use a `Vec`?
```suggestion
) -> Vec<AnyClassProperties> {
```